dotCMS/src/main/java/com/dotmarketing/image/filter/SubSampleImageFilter.java

```diff
@@ -0,0 +1,61 @@
+package com.dotmarketing.image.filter;
+
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.util.Map;
+import javax.imageio.ImageIO;
+import com.dotmarketing.exception.DotRuntimeException;
+
+public class SubSampleImageFilter extends ImageFilter {
+
+
+    public String[] getAcceptedParameters() {
+        return new String[] {"w (int) specifies width", "h (int) specifies height",};
+    }
+
+    public File runFilter(final File file, Map<String, String[]> parameters) {
+        int width = parameters.get(getPrefix() + "w") != null ? Integer.parseInt(parameters.get(getPrefix() + "w")[0])
```

Comment:
I'd cache whatever `parameters.get(getPrefix() + "w")` (and `parameters.get(getPrefix() + "h")`) in a couple of variables.

Reply:
Done